libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -522,28 +532,28 @@ void AC_PosControl::init_xy()
 
 /// input_accel_xy - calculate a jerk limited path from the current position, velocity and acceleration to an input acceleration.
 ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
-///     The kinematic path is constrained by the maximum acceleration and time constant set using the function set_max_speed_accel_xy and time constant.
-///     The time constant defines the acceleration error decay in the kinematic path as the system approaches constant acceleration.
-///     The time constant also defines the time taken to achieve the maximum acceleration.
+///     The kinematic path is constrained by the maximum acceleration and jerk set using the function set_max_speed_accel_xy.
+///     The jerk limit defines the acceleration error decay in the kinematic path as the system approaches constant acceleration.
+///     The jerk limit also defines the time taken to achieve the maximum acceleration.
 void AC_PosControl::input_accel_xy(const Vector3f& accel)
 {
     // check for ekf xy position reset
     handle_ekf_xy_reset();
 
     update_pos_vel_accel_xy(_pos_target.xy(), _vel_desired.xy(), _accel_desired.xy(), _dt, _limit_vector.xy());
-    shape_accel_xy(accel, _accel_desired, _accel_max_xy_cmss, _tc_xy_s, _dt);
+    shape_accel_xy(accel, _accel_desired, _jerk_xy_max, _dt);
 }
 
 /// input_vel_accel_xy - calculate a jerk limited path from the current position, velocity and acceleration to an input velocity and acceleration.
 ///     The vel is projected forwards in time based on a time step of dt and acceleration accel.
 ///     The function takes the current position, velocity, and acceleration and calculates the required jerk limited adjustment to the acceleration for the next time dt.
-///     The kinematic path is constrained by the maximum acceleration and time constant set using the function set_max_speed_accel_xy and time constant.
-void AC_PosControl::input_vel_accel_xy(Vector2f& vel, const Vector2f& accel)
+///     The kinematic path is constrained by the maximum acceleration and jerk set using the function set_max_speed_accel_xy.
+void AC_PosControl::input_vel_accel_xy(Vector2f& vel, const Vector2f& accel, bool limit_output)
```

Comment:
Could we add a comment explaining when the "limit_output" argument should be set?  I guess there are a few methods which take this new argument so we probably need to add this comment to both the .h and .cpp in at least two places.

Reply:
I am adding:

///     The parameter limit_output specifies if the velocity and acceleration limits are applied to the sum of commanded and correction values or just correction.